internal/cli/completion.go

```diff
@@ -174,7 +174,7 @@ func (kna KindNameAutocomplete) ValidArgsFunction(cmd *cobra.Command, args []str
 
 	if len(args) == 0 {
 		kindLike, _, _ := strings.Cut(toComplete, "/")
-		if kind, err := ResourceKindFromString(kindLike); err == nil {
+		if kind, err := ResourceKindFromString(kindLike); err == nil && slices.Contains(kna.AllowedKinds, kind) {
```

Comment:
optional this would be a nice helper

```suggestion
		if kind, err := ResourceKindFromString(kindLike); err == nil && kna.isKindAllowed(kind) {
```